docs/spec/store/interblock-cache.md

```diff
@@ -0,0 +1,293 @@
+# Inter-block Cache
+
+- [Inter-block cache](#inter-block-cache)
+  - [Synopsis](#synopsis)
+  - [Overview and basic concepts](#overview-and-basic-concepts)
+    - [Motivation](#motivation)
+    - [Definitions](#definitions)
+  - [System model and properties](#system-model-and-properties)
+    - [Assumptions](#assumptions)
+    - [Properties](#properties)
+      - [Thread safety](#thread-safety)
+      - [Crash recovery](#crash-recovery)
+      - [Iteration](#iteration)
+  - [Technical specification](#technical-specification)
+    - [General design](#general-design)
+    - [API](#api)
+      - [CommitKVCacheManager](#commitkvcachemanager)
+      - [CommitKVStoreCache](#commitkvstorecache)
+    - [Implementation details](#implementation-details)
+  - [History](#history)
+  - [Copyright](#copyright)
+
+
+## Synopsis
+
+The inter-block cache is an in-memory cache storing (in-most-cases) immutable state that modules need to read in between blocks. When enabled, all sub-stores of a multi store, e.g., `rootmulti`, are wrapped.
```

Comment:
can we drop (in-most-cases)?

```suggestion
The inter-block cache is an in-memory cache storing immutable state that modules need to read in between blocks. When enabled, all sub-stores of a multi store, e.g., `rootmulti`, are wrapped.
```

Reply:
Actually, I wonder if there is any guarantee that the inter-block cache stores the immutable state. Opinions on what enables this? Is it the ARC? If so, I will highlight this.

Reply:
It's not immutable. If a key is updated, the cache will be updated.

Reply:
I know it is not immutable. My point is whether there is anything in place that in case of reaching the max capacity of the cache, it is likely that immutable state remains cache. My guess is that the fact that the cache is an ARC may help with that: it tracks both frequency and recency of use. Thus, under the assumption that immutable state is more frequently queried, the ARC may help guaranteeing that this is almost always cached, even when max capacity is reached. 

If we agree that the above is correct, I will highlight that it is important that the cache implementation is an ARC (or something similar that enables the above), instead of just discussing it as an implementation details.

Also, I want to get rid of the word immutable, it is confusing: nothing is immutable per se, we only mean keys that a rarely updated.